### DISCUSSION TITLE
Manually Trigger Motion Event

### DISCUSSION THREAD

**Discussion by luisiam** [roflcoopter/viseron#726] [Ideas]
Is it possible to trigger motion detection event via some sort of API or MQTT? Afterwards, viseron will trigger any AI detection that's configured. The purpose is that modern cameras have pretty good motion detection and we can offload the motion detection load to the cameras instead of having a consistent load on the server.

**Comment by roflcoopter** [comment]
Not possible right now but it is on the to do list!
I have already laid some ground work for this by separating out the frame scanning logic to a separate class: https://github.com/roflcoopter/viseron/blob/dev/viseron/domains/motion_detector/__init__.py#L267

So adding the support for an external trigger will be trivial.

Just need to finish up the 24/7 recordings first which is getting close!

Converting this to an issue so i can track it properly.

**Comment by roflcoopter** [comment]
This is now implemented in `dev` through the MQTT component, will be in the next release
